src/index.js

```diff
@@ -176,26 +206,70 @@ export class BlockSuggestor {
   }
 }
 
+
+
 /**
  * Main entry point to initialize the suggested blocks categories.
- * @param {Blockly.WorkspaceSvg} workspace the workspace to load into
+ * @param {Blockly.WorkspaceSvg|Element|string} workspaceOrContainer the workspace to load into WorkspaceSvg, or an Element, or string
  * @param {number} numBlocksPerCategory how many blocks should be included per
  * category. Defaults to 10.
  * @param {boolean} waitForFinishedLoading whether to wait until we hear the
  * FINISHED_LOADING event before responding to BLOCK_CREATE events. Set to false
  * if you disable events during initial load. Defaults to true.
  */
 export const init = function (
-  workspace,
+  workspaceOrContainer,
   numBlocksPerCategory = 10,
   waitForFinishedLoading = true,
 ) {
+  // Variable to store the Blockly workspace instance
+  // Variable to store the container element
+  let workspace
+  let container;
   const suggestor = new BlockSuggestor(numBlocksPerCategory);
+  // Check if 'workspaceOrContainer' is already a Blockly workspace
+  // If 'workspaceOrContainer' is a string, attempt to find the container element
+  // If 'workspaceOrContainer' is not a string, treat it as a container element
+  if (workspaceOrContainer instanceof Blockly.WorkspaceSvg) {
+    workspace = workspaceOrContainer;
+  }
+  else {
+      if (typeof workspaceOrContainer === 'string') {
+        container =
+        document.getElementById(workspaceOrContainer) || document.querySelector(workspaceOrContainer);
+      }
+      else{
+        container = workspaceOrContainer;
+    }
```

Comment:
Use the [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator) when assigning `container` to avoid the need for the `if/else`.